jetty-core/jetty-session/src/main/java/org/eclipse/jetty/session/AbstractSessionManager.java

```diff
@@ -244,6 +239,11 @@ public void doStart() throws Exception
         _context = ContextHandler.getCurrentContext();
         _loader = Thread.currentThread().getContextClassLoader();
 
+        // ensure a sessiun path is set for non root contexts
+        String contextPath = _context == null ? "/" : _context.getContextPath();
+        if (!"/".equals(contextPath) && getSessionPath() == null)
+            setSessionPath(contextPath);
```

Comment:
This seems unrelated to this HttpCookie effort

Reply:
This was part of cleaning up the session cookie configuration.  This code was moved from every time a session cookie was created to here in doStart